src/Console/Report/FixReport/ReportSummary.php

```diff
@@ -21,6 +21,9 @@
  */
 final class ReportSummary
 {
+    /**
+     * @var array<string, array{appliedFixers: list<string>, diff?: string}>
```

Comment:
In `Runner::fixFile()` return type the `diff` key is not optional, but there are tests for `ReportSummary` with no diff, and some reporters have `empty($fixResult['diff'])` check. Can diff be empty under some circumstances, or is it a residue from earlier versions?

Reply:
the code-diff generation is optional. with that, you may not assume that it's always present.

what does it mean for types? I'm not sure if we return `null` or `empty string` when we do not generate diff

Reply:
We have a `NullDiffer`, which returns an empty string, so probably that's the case.

I'll update optional types and add an empty string to the tests where it's missing.